src/main/java/org/spongepowered/common/resourcepack/SpongeWorldResourcePack.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.resourcepack;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URLEncoder;
+
+public class SpongeWorldResourcePack extends SpongeResourcePack {
+
+    private final String path;
+
+    public SpongeWorldResourcePack(String levelUrl, String hash) {
+        super(hash);
+        this.path = levelUrl.substring("level://".length());
+    }
+
+    @Override
+    public String getName() {
+        return "resourceszip";
+    }
+
+    @Override
+    public String getUrlString() {
+        return "level://" + this.path;
+    }
+
+    @Override
+    public URI getUri() {
+        try {
+            return new URI("level://" + URLEncoder.encode(path, "UTF-8"));
```

Comment:
Instead of using the string ctor, maybe use `URI.create` factory method instead, since you're just catching this exception further up the call hierarchy and wrapping it in an `IllegalArgumentException` anyway, which is what `URI.create` does.